### PR TITLE
Debug interface

### PR DESCRIPTION
**Pull request by tballmsft**

### CODE REVIEW

File: src/vm/vmrunner.ts [hunk 1]
```diff
@@ -460,6 +496,28 @@ export class VMProgramRunner extends JDClient {
         return this._watch[sourceId]
     }
 
+    // breakpoints
+    setBreakpoints(breaks: string[]) {
+        breaks.forEach(b => {
```
Comment (pelikhan) [inline src/vm/vmrunner.ts:501]: should you clear previous breakpoints here too?

File: src/vm/vmrunner.ts [hunk 2]
```diff
@@ -460,6 +496,28 @@ export class VMProgramRunner extends JDClient {
...
+            this._breaks = {}
+        } else {
+            breaks.forEach(b => {
+                delete this._breaks[b]
+            })
+        }
+    }
+
+    breakPointOn(id: string) {
+        return id in this._breaks
```
Comment (pelikhan) [inline src/vm/vmrunner.ts:517]: ``return this._breaks?.[id]``

Comment (tballmsft) [reply src/vm/vmrunner.ts:516]: what is !! in Javascript?

File: src/vm/vmrunner.ts (hunk 1 above)
Comment (tballmsft) [reply src/vm/vmrunner.ts:500]: OK, so you will always send me the full set of breakpoints from Blockly (no incremental shit here)?

**Pull request merged**